File: src/server/src/velvet_dawn/mechanics/selectors/selector_self.py
from typing import List

from .selector import Selector
from ...db.instances import Instance

""" Selector 'self' modifies what ever is calling it

Some examples:
 - {"targets": "self", "set": "civil-war:okay"}
 - {"if": "self", "equals": "civil-war:me"}
 - {"targets": "self.movement.weight", "set": 5}
 - {"if": "self.custom", "equals": 8}
 - {"if": "self", "equals": "civil-war:me"}
 - {"targets": "self.movement.weight", "set": 5}
 - {"targets": "self.health.max-health", "set": 5}
 - {"if": "self.attributes.custom", "equals": 8}
 - {"if": "self.combat.range", "equals": 8}
"""


class SelectorSelf(Selector):
    def __init__(self):
        Selector.__init__(
            self,
            selector_name="self"
        )

    def new(self):
        return SelectorSelf()

    def get_selection(self, instance: Instance) -> List[Instance]:
        return self.filters.filter(instance, [instance])

File: src/jserver/src/velvetdawn/mechanics/selectors/SelectorSelf.java
package velvetdawn.mechanics.selectors;

import velvetdawn.VelvetDawn;
import velvetdawn.models.instances.Instance;

import java.util.Collection;
import java.util.List;

public class SelectorSelf extends Selector {

    /* Return the instance calling the selector */

    public SelectorSelf(VelvetDawn velvetDawn) {
        super(velvetDawn);
    }

    @Override
    protected Collection<Instance> getSelection(Instance instance) {
        return this.filters.filter(instance, List.of(instance));
    }
}
